service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -43,27 +36,35 @@ public class DatasetService {
   private final SamService samService;
   private final DatasetDao datasetDao;
   private final ObjectMapper objectMapper;
+  private final StorageSystemService externalService;
 
   private static final int MAX_ROWS = 30;
 
-  @Autowired
   public DatasetService(
       DatarepoService datarepoService,
       RawlsService rawlsService,
+      ExternalSystemService externalService,
       SamService samService,
       DatasetDao datasetDao,
       ObjectMapper objectMapper) {
     this.datarepoService = datarepoService;
     this.rawlsService = rawlsService;
+    this.externalService = externalService;
     this.samService = samService;
     this.datasetDao = datasetDao;
     this.objectMapper = objectMapper;
   }
 
-  public static class IllegalMetadataException extends RuntimeException {
-    public IllegalMetadataException(Throwable cause) {
-      super(cause);
-    }
+  private StorageSystemService getService(StorageSystem system) {
+    return switch (system) {
```

Comment:
I'd like to get this out of this class but didn't see an easy way to do it, since enums instances are static and don't directly interact with spring injected objects. Maybe something for a future refactor.